Collect questions that we would like to answer (#43)
comit-network/xmr-btc-swap: docs/research/user-experience/xmr-btc-user-interface-validation.md

# XMR<>BTC Atomic Swap - User Interface Prototype Validation

This document:

1. Collects the validation criteria.
2. Lists the created user interface prototypes, and link to Figma.
3. Maps the protoypes to the validation criteria.

This document will be updated with new information during the course of the project.

## Questions

The questions are split between `M`aker (liquidity provider) and `T`aker (normal user), because the objectives are somewhat different.

|  **Topic** | **High Level Questions** | **More specific question** | **User is happy to...** | **Actor** |
| --- | --- | --- | --- | --- |

> |  Node & Wallet Management | How do users monitor the Bitcoin and Monero blockchain? | Is a user fine with trusting a third party to monitor transactions? | use a service like blockchain.com to retrieve blocks for validation | TM |
> |   |  |  | run his own Bitcoin node, third party service for Monero | TM |
> |   |  |  | run his own Monero node, third party service for Bitcoin | TM |
> |   |  |  | run both his own Bitcoin and Monero node | TM |
> |   | How do users brodcast transactions to Bitcoin and Monero? | Is a user fine with trusting a third party to broadcast transactions? | use a wallet that connects to third party nodes | TM |
> |   |  |  | send signed transactions through third part nodes | TM |
> |   |  |  | run his own Bitcoin node, third party service for Monero | TM |
> |   |  |  | run his own Monero node, third party service for Bitcoin | TM |
> |   |  |  | run both his own Bitcoin and Monero node | TM |

Reviewer: ``````suggestion
   |  Node & Wallet Management | How do users monitor the Bitcoin and Monero blockchains? | Is a user fine with trusting a third party to monitor transactions? | use a third party service to retrieve transactions | TM |
   |   |  |  | run his own blockchain node | TM |
   |   | How do users brodcast transactions to Bitcoin and Monero? | Is a user fine with trusting a third party to broadcast transactions? | use a wallet that connects to third party nodes | TM |
   |   |  |  | send signed transactions through third part nodes | TM |
   |   |  |  | run his own full blockchain node | TM |
   |   |  |  | run a SPV node (Bitcoin) | TM |
   ```

IMHO, This adds quite a bit of text with little added value. I was not going to comment as I thought I was nitpicking but then I saw the same pattern for the wallet and think it is a bit too much.

I think it's clear enough in the way I propose. Feel free to disagree 🙃

Author: I was not sure about this one myself - I opted for distinguishing the sides, because users might actually consider that.
Reasoning: A monero user already has Monero set up but might not care about setting up and running a Bitcoin full node. Since I am just not sure if that applies or not I thought I keep it separate for now, so it can be validated that users are willing to run both nodes.

Adding the SPV node is an interesting option! It shows, once again, that there are quite a lot of options, and making them mutually exclusive is hard 😬

Author: I'll adapt it to what you propose! We just have to keep in mind to record the user answers for both sides. I would reckon that a person that cares about running a Monero full node also cares about running a bitcoin full node, but when it e.g. comes to Ethereum that is not true. Most people would trust Infura for that.

{}-A ... Prototype for product A (single market maker)
{}-B ... Prototype for product B (decentralized trading platform)
```

Example:

`D-A2-1`: Mock showing detail 1 for prototype `P-A1`

### Figma Links

* [P-A1](https://www.figma.com/proto/QdvmbRYuBpEpFI3D1R4qyM/XMR-BTC_SingleMaker_LowFidelity?node-id=54%3A4894&viewport=1503%2C-52%2C0.5576764941215515&scaling=min-zoom): Webpage for discovery, CLI for execution
* [P-A2](https://www.figma.com/proto/QdvmbRYuBpEpFI3D1R4qyM/XMR-BTC_SingleMaker_LowFidelity?node-id=7%3A4377&viewport=696%2C-250%2C0.362735778093338&scaling=min-zoom): Webpage for discovery, GUI for execution
* [D-A2-1](https://www.figma.com/proto/QdvmbRYuBpEpFI3D1R4qyM/XMR-BTC_SingleMaker_LowFidelity?node-id=235%3A1374&viewport=1336%2C-1825%2C0.7878535389900208&scaling=min-zoom): GUI swap execution steps for `send` `BTC`, `receive` `XMR`
* [D-A2-2](https://www.figma.com/proto/QdvmbRYuBpEpFI3D1R4qyM/XMR-BTC_SingleMaker_LowFidelity?node-id=128%3A8016&viewport=1404%2C-1158%2C0.66261225938797&scaling=min-zoom): GUI swap execution steps for `send` `XMR`, `receive` `BTC`

### Mapping of Prototype to validation criteria

> |  **User is happy to...** | **Actor** | **P-A1** | **P-A2** | **D-A2-1** | **D-A2-2** |

Reviewer: I would suggest to merge both tables to make our life easier and not have to ensure they are in sync.
I am happy to have a crack at it if you already tried and do not think there is an elegant way to achieve that.

Author: It actually is one table in the google sheet where I manage this - I added you.
My plan was to manage the data in the gsheet, and once it reaches a state where it adds value to publish it update both tables in this document.
I use a gsheet plugin to generate the markdown tables, only takes a few seconds to generate the code and then it's copy paste. I know it's not optimal to have the data in a gsheet and here, but it's just way easier to fill and manage this in a gsheet.

Author: I kept the two table approach for now. If we produce a high number of prototypes it will be hard to keep adding columns to one complete markdown table. If you feel strong about this please ping me.

Reviewer: I am not going to tell you how to do it :) It was merely a suggestion to make your live easier ;)

# XMR<>BTC Atomic Swap - User Interface Prototype Validation

This document sums up the questions that we would like to validated with mocks.

## Questions

The questions are split between maker (liquidity provider) and taker (normal user), because the objectives are somewhat different.

> Is a `taker` happy to...

Reviewer 2: I think the points cover pretty much what we want to find answers to. 

A thought when reading them: 
At the end it boils down to how you ask the user the question. The answer to `Are you happy running a bitcoind full-node` will be different to `Would you trust a third party with monitoring the blockchain for you`. 

It might help us to change our own mindset if we phrase these points differently in from of questions to ourselves e.g. 

* `How does a user/taker/maker currently broadcast/monitor bitcoin transactions?`
  * `Is a user fine with trusting a third party such as a block explore (blockchain.com/blockstream.com)?` 
* `How does a user/taker/maker hide its identity (if at all)?`

If you want, I can help you rephrasing some of the points.

Reviewer: > I think the points cover pretty much what we want to find answers to.
> 
> A thought when reading them:
> At the end it boils down to how you ask the user the question. The answer to `Are you happy running a bitcoind full-node` will be different to `Would you trust a third party with monitoring the blockchain for you`.

Just so we are clear, we are not asking a user directly.
At this stage, we assume usage of full node. If we were to move into a direction of not requiring full node, most likely because the user uses another wallet, then yes indeed we would have to find a way to show the user that we are sending data to an blockchain explorer. Maybe via a note or maybe letting the user configure which explorer.

> It might help us to change our own mindset if we phrase these points differently in from of questions to ourselves e.g.
> 
>     * `How does a user/taker/maker currently broadcast/monitor bitcoin transactions?`

I agree, we should add a question `What bitcoin/monero wallet does the user want to use to swap?`

@da-kami maybe the "is happy" format to restrictive to properly phrase the questions we want to answer.

>       
>       * `Is a user fine with trusting a third party such as a block explore (blockchain.com/blockstream.com)?`

as stated above, I am not sure this is relevant at this stage.

>     * `How does a user/taker/maker hide its identity (if at all)?`

This question is very broad and I believe make it hard to validate in a mock.
I prefer to see `Is the user happy to contact other party using clear ip?` so that we know the answer is yes or no.

> 
> If you want, I can help you rephrasing some of the points.

We are not sure to understand what your point is. These are not interview questions and you seem to think they are by the way you started your reply. Can you please clarify a bit more what issue you see with the current format of the questions?

# XMR<>BTC Atomic Swap - User Interface Prototype Validation

This document sums up the questions that we would like to validated with 

## Questions

With the UI prototypes we are trying to answer the following questions.
The questions are split between maker and taker, because the objectives are somewhat different.

Is a `taker` happy to...

1. run a bitcoind full-node
1. run a monerod full-node
1. use a webpage for discovering makers
1. use a privacy chat tool for discovering makers

> 1. manually add makers to retrieve orders

Author: I would not rule it out - this would be the most simple discovery. 
What I was planning to validate is, if users care a lot about having automated discovery. Some might actually prefer using their current privacy preserving tools and then just copy paste. 
Not sure this is relevant though, given what you commented here: https://github.com/comit-network/xmr-btc-swap/pull/43#discussion_r527256451

> 1. manually add makers to retrieve orders

Author: removed it.

1. download software (swap execution daemon) before being able to do a swap

> 1. run the swap daemon locally on the taker's machine

Author: removing that one, because it is included in the next one 
`keep a GUI/CLI open for the whole length of the swap execution`

1. keep the swap daemon running for the whole length of the swap execution
1. give control over the bitcoin and monero wallet to the swap execution daemon
1. keep the browser open for the whole length of the swap execution
1. be in different cryptographic roles depending on the direction of the swap

> 1. run the swap execution daemon without using Tor
> 1. connect to a maker that does not use Tor

Reviewer: ```suggestion
1. to not use tor
```

I would not separate both.

Makes me think, in the mocks, we need to be sure to add an address for the maker on both CLI and swap UI (and this address can be ip4 to validate this assumption).

Author: Yes! I'll adapt what is currently there to show ip4 addresses. 

Note:
In the long run we might want use more, simple mocks that focus on specific questions rather than one big flow. I don't think we can expect users to focus on too many small details. Alternatively I could also add more stickies to point out question points. 
Not sure what is the best way there yet.

... to do an XMR<>BTC swap?

Is a `maker` happy to...

1. run a bitcoind full-node
1. run a monerod full-node
1. set up a webpage where takers get a price 
1. publicly advertise being a maker

> 1. be in different cryptographic roles depending on the direction of the swap

Reviewer: same than for taker
